pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -926,7 +925,7 @@ public void connectionFailed(PulsarClientException exception) {
         } else {
             ByteBuf cmd = Commands.newCloseConsumer(consumerId, requestId);
             cnx.sendRequestWithId(cmd, requestId).handle((v, exception) -> {
-                boolean ignoreException = !cnx.ctx().channel().isActive();
+                boolean ignoreException = cnx.ctx() == null || !cnx.ctx().channel().isActive();
```

Comment:
it looks like this is the fix.

if the result of` cnx.ctx()` may be different from one call to the second call I suggest you to capture the value once

```
ctx = cnx.ctx();
boolean ignoreException = ctx == null || !ctx.channel().isActive()
```

Reply:
You are definitely right! Thank you very much.